packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/component_refs.rs

```diff
@@ -0,0 +1,40 @@
+use super::content_refs::ContentRef;
+use crate::components::types::ComponentIdx;
+use anyhow::anyhow;
+
+/// A vector of references to components
+///
+/// TODO: when have array props, turn this into just `ComponentRef`, which is a reference to a single component
+/// and then use the general array prop mechanism to refer to multiple components
+#[derive(Debug, Clone, Default, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct ComponentRefs(pub Vec<ComponentIdx>);
+
+/// A reference to a single component
+#[derive(Debug, Clone, Default, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct ComponentRef(pub ComponentIdx);
+
+impl ComponentRef {
+    pub fn as_content_ref(&self) -> ContentRef {
```

Comment:
This is identical to the `impl From<ComponentRef> for ContentRef` in `content_refs.rs`, presumably created as a convenience for cases such as in `SerialNumberProp`. Just curious for rationale for knowing when to create something like `as_content_ref()` rather than using `into()`.